src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentExcerptService.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// excerpt some part of <see cref="Document"/>
```

Comment:
didn't want to tie this to actual razor case since I want xaml to keep thier current design but still use this. for example, they can still keep current g.i.cs file thing they generates, but use this service to show right thing on FAR or code lens.